source/trigger/iterator.go

```diff
@@ -0,0 +1,477 @@
+	queryTriggerCreate = `
+		CREATE OR REPLACE
+		TRIGGER %s
+		AFTER INSERT OR UPDATE OR DELETE ON %s
+		FOR EACH ROW EXECUTE PROCEDURE %s();`
+	queryTriggerDrop       = "DROP TRIGGER IF EXISTS %s;"
+	queryColumnNamesSelect = "SELECT column_name FROM information_schema.columns WHERE table_name = $1;"
+	queryPrimaryKeySelect  = `
+		SELECT column_name
+		FROM information_schema.key_column_usage
+		WHERE table_name = $1 AND constraint_name LIKE '%_pkey'
+		LIMIT 1;`
+)
+
+// declare Postgres $ placeholder format
+var psql = sq.StatementBuilder.PlaceholderFormat(sq.Dollar)
+
+// Iterator is an implementation of an iterator in a trigger CDC mode.
+type Iterator struct {
+	conn     *pgxpool.Pool
+	position *Position
+	snapshot *Snapshot
+	cdc      *CDC
+
+	// table is a table's name.
+	table string
+	// orderingColumn is a name of column what iterator use for sorting data.
+	orderingColumn string
+	// key is a column's name that iterator will use for setting key in record.
+	key string
+	// columns is a list of table's columns for record payload.
+	// if empty - will get all columns.
+	columns []string
+	// batchSize is a size of batch.
+	batchSize uint64
+	// conduit is names of conduit items in a DB (like tracking table, trigger, and function).
+	conduit string
+}
+
+// Params is an incoming iterator params for the New function.
+type Params struct {
+	Pos            sdk.Position
+	Conn           *pgxpool.Pool
+	Table          string
+	OrderingColumn string
+	Key            string
+	Snapshot       bool
+	Columns        []string
+	BatchSize      uint64
+}
+
+// New creates a new instance of the iterator.
+func New(ctx context.Context, params Params) (*Iterator, error) {
+	pos, err := ParseSDKPosition(params.Pos)
+	if err != nil {
+		return nil, fmt.Errorf("parse position: %w", err)
+	}
+
+	iterator := &Iterator{
+		conn:           params.Conn,
+		position:       pos,
+		table:          params.Table,
+		orderingColumn: params.OrderingColumn,
+		key:            params.Key,
+		columns:        params.Columns,
+		batchSize:      params.BatchSize,
+		conduit:        fmt.Sprintf("conduit_%s_%s", params.Table, pos.CreatedAt),
+	}
+
+	err = iterator.populateKey(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("populate key: %w", err)
+	}
+
+	err = iterator.populateColumns(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("populate names of the table columns: %w", err)
+	}
+
+	err = iterator.setupCDC(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("setup cdc: %w", err)
+	}
+
+	if params.Snapshot && pos.Mode == ModeSnapshot {
+		iterator.snapshot, err = NewSnapshot(ctx, SnapshotParams{
+			Conn:           iterator.conn,
+			Position:       iterator.position,
+			Table:          iterator.table,
+			OrderingColumn: iterator.orderingColumn,
+			Key:            iterator.key,
+			Columns:        iterator.columns,
+			BatchSize:      iterator.batchSize,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("init snapshot iterator: %w", err)
+		}
+	} else {
```

Comment:
I would rather avoid the `else` by converting the `if/else` statement to a `switch-case`.

Reply:
I agree with you.
It has been simplified.